### PR TITLE
feat(server): use wildcard search for originalFileName and originalPath

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
* support wildcard search for `originalPath` (relevant for external libraries with custom folder structures)
* don't add wildcard before and after the search term for `originalFileName`
* replace `*` with `%` for use with SQL `ILIKE`

Should we add a section for path in the search form, next to context and filename?

### CODE REVIEW

File: server/src/utils/database.ts [hunk 1]
```diff
@@ -58,12 +58,18 @@ export function searchAssetBuilder(
...
   builder.andWhere(_.omitBy(path, _.isUndefined));
 
   if (options.originalFileName) {
     builder.andWhere(`f_unaccent(${builder.alias}.originalFileName) ILIKE f_unaccent(:originalFileName)`, {
-      originalFileName: `%${options.originalFileName}%`,
+      originalFileName: options.originalFileName.replaceAll('*', '%'),
+    });
+  }
+
+  if (options.originalPath) {
```
Comment (alextran1502) [inline server/src/utils/database.ts:70]: I don't think we have this in the UI

Comment (andreasgerstmayr) [reply server/src/utils/database.ts:70]: Not yet, should I add it? It's relevant for existing folder structures when using external libraries.

Comment (jrasm91) [reply server/src/utils/database.ts:70]: I don't think this has an index so it probably won't work very well as is.

Comment (mertalev) [reply server/src/utils/database.ts:70]: Is there a need to handle path and filename separately? It seems to me that using `originalPath` would work for both.

Comment (jrasm91) [reply server/src/utils/database.ts:70]: For non-external assets they are different. Or, depending on you storage template they can be completely independent.

Comment (andreasgerstmayr) [reply server/src/utils/database.ts:70]: > No, we plan to offer these as advanced free text searches in which we include some type of DSL in the input text. So probably it makes sense to have us implement this in the future altogether

A single text input where I can enter something like `path:"*/Paris 2022/*"`? That sounds great, I guess I can close this ticket then.

**Pull request closed without merging**